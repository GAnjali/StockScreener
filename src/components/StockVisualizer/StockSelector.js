import React from 'react';
import Select from 'react-select';

const StockSelector = (props) => {
    let stocks = props.stocks;
    let selectorOptions = stocks.map((data) => {
            return {value: data, label: data}
        }
    );

    return (
        <Select id="stock-selector"
                options={selectorOptions}
                onChange={(selectedOption) => {
                    props.onSelectChange(selectedOption.value);
                }}
                theme={(theme) => ({
                    ...theme,
                    borderRadius: 0,
                    boxShadow: 2,
                    colors: {
                        ...theme.colors,
                        text: 'orangered',
                        primary25: '#d1cbcb',
                        primary: '#d1cbcb',
                        primary50: '#d1cbcb'
                    }
                })}
        />
    )
};

export default StockSelector;