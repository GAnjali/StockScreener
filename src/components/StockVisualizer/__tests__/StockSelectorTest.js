import React from "react";
import {shallow} from "enzyme";
import StockSelector from "../StockSelector";

describe("StockSelector Tests", () => {
    it("Should render without crashing", () => {
        shallow(<StockSelector stocks={[]}/>);
    });

    it('should match the Home snapshot', function () {
        const stockSelectorComponent = shallow(<StockSelector stocks={[]}/>);
        expect(stockSelectorComponent).toMatchSnapshot();
    });
});
