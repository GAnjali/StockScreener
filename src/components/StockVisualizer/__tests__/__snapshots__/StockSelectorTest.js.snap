// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockSelector Tests should match the Home snapshot 1`] = `ShallowWrapper {}`;
